File: synapse-runtime/common/models/queue/queue_job.py
from datetime import datetime
from typing import Generic, TypeVar

from pydantic import BaseModel, Field

from common.enums.job_type import JobType

T = TypeVar("T")


class QueueJob(BaseModel, Generic[T]):
    timestamp: datetime
    message: str
    job_type: JobType = Field(alias="jobType")
    payload: T

File: synapse-transport/src/main/java/com/synapse/transport/exception/dto/QueueJob.java
package com.synapse.transport.exception.dto;

import com.synapse.transport.common.enums.JobType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QueueJob<T> {
    private String timestamp;
    private String message;
    private JobType jobType;
    private T payload;
}
